Detectors/TRD/base/test/testCoordinateTransforms.cxx

```diff
@@ -0,0 +1,161 @@
+{
+namespace trd
+{
+
+using namespace o2::trd::constants;
+using namespace std;
+
+void testRCPoint(double calculatedPoint, double predictedPoint)
+{
+  float e = 0.000001;
+  BOOST_CHECK(fabs(predictedPoint - calculatedPoint) <= e);
+}
+
+BOOST_AUTO_TEST_CASE(LocaltoRCTest)
+{
+  auto mGeo = o2::trd::Geometry::instance();
+  mGeo->createPadPlaneArray();
+  mGeo->createClusterMatrixArray();
+
+  int hcid = 776;
+  // This C1 chamber has 16 pad rows with I pad length = 90mm and O pad length = 75mm
+  int detector = hcid / 2;
+  int stack = mGeo->getStack(detector);
+  int layer = mGeo->getLayer(detector);
+
+  auto padPlane = mGeo->getPadPlane(layer, stack);
+  double lengthIPad = padPlane->getLengthIPad();
+  double lengthOPad = padPlane->getLengthOPad();
+
+  double padIWidth = padPlane->getWidthIPad();
+  double padOWidth = padPlane->getWidthOPad();
+
+  double tiltingAngle = padPlane->getTiltingAngle();
+
+  // Test padrows
+  auto p1 = padPlane->getPadRow(0);
+  // Center of the chamber. This should return the lower edge of padrow 8.
+  // Since we are using float values, the lower edge of padrow 8 correspond with float value 8.0.
+  testRCPoint(p1, 8.);
+
+  auto p2 = padPlane->getPadRow(lengthIPad / 2.);
+  // With an I pad length of 9 cm, 9. / 2. should put us half way into the preceeding pad (pad 7) since padrow number
+  // decreses in positive z direction.
+  testRCPoint(p2, 7.5);
+
+  auto p3 = padPlane->getPadRow(-lengthIPad / 2.);
+  // Same as above but in the other direction.
+  testRCPoint(p3, 8.5);
+
+  auto p4 = padPlane->getPadRow(lengthIPad * 4.2);
+  // Arbitrary distance in z.
+  testRCPoint(p4, 8 - 4.2);
+
+  auto p5 = padPlane->getPadRow(lengthIPad * 7 + lengthOPad);
+  // Lower border case. Take center and add 7 pads * 9 cm + 1 pad * 7.5 cm.
+  testRCPoint(p5, 0);
+
+  auto p6 = padPlane->getPadRow(-lengthIPad * 7 - lengthOPad);
+  // Upper border case. Take center and subtract 7 pads * 9 cm - 1 pad * 7.5 cm.
+  testRCPoint(p6, 16);
+
+  // Test pads without pad tilting
+  // auto p7 = padPlane->getPad(0);
+  // // Center of the chamber. This should return the lower edge of pad 72.
+  // // Since we are using float values, the lower edge of pad 72 correspond with float value of 72.0.
+  // testRCPoint(p7, 72);
+
+  // auto p8 = padPlane->getPad(padIWidth / 2.);
+  // // Move forward 1/2 of inner pad width. Pad number increases in positive y direction so should get 72.5
+  // testRCPoint(p8, 72.5);
+
+  // auto p9 = padPlane->getPad(-padIWidth / 2.);
+  // // Move backward 1/2 inner pad width
+  // testRCPoint(p9, 71.5);
+
+  // auto p10 = padPlane->getPad(padIWidth * 7.77);
+  // // Move an arbitrary distance in y
+  // testRCPoint(p10, 72 + 7.77);
+
+  // auto p11 = padPlane->getPad(padIWidth * 71 + padOWidth);
+  // // Top border case right on the edge of the chamber
+  // testRCPoint(p11, 144);
+
+  // auto p12 = padPlane->getPad(-padIWidth * 71 - padOWidth);
+  // // Lower border case
+  // testRCPoint(p12, 0);
```

Comment:
What's the purpose of these commented lines?

Reply:
Thanks so much for all the help! These lines were the initial testing I did before adding pad tilting into the calculations. It seemed like a good idea to start with the simpler problem first and ensure that was working correctly before progressing to the more complex problem involving pad tilting. I just left it there in case I ever needed it again, but I guess this is unnecessary at this point.